src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -47,7 +50,13 @@ private void Workspace_DocumentClosed(object sender, DocumentEventArgs e)
 
                     // Ensure we dispose the context, which we'll do asynchronously
                     contextTask.ContinueWith(
-                        t => t.Result.Dispose(),
+                        t =>
+                        {
+                            var context = t.Result;
+
+                            context.CodingConventionsChangedAsync -= OnCodingConventionsChangedAsync;
```

Comment:
❓ Does this matter if we are disposing of the object anyway?

Reply:
This is an editor's component, but please be a good citizen still :)
I wonder if you need to unsubscribe before scheduling a task though, what if the .editorconfig file has changed in between lines 52 and 57 and OnCodingConventionsChangedAsync gets called?